extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php

```diff
@@ -43,6 +43,30 @@ public static function replaceInfoboxMarkers( &$parser, &$text ) {
 	}
 
 	/**
+	 * @param $markup
+	 * @param Parser $parser
+	 * @param PPFrame $frame
+	 * @return string
+	 * @throws UnimplementedNodeException when node used in markup does not exists
+	 * @throws XmlMarkupParseErrorException xml not well formatted
+	 */
+	public function render( $markup, Parser $parser, PPFrame $frame, $params = null ) {
+		$infoboxParser = new Wikia\PortableInfobox\Parser\XmlParser( $this->getFrameParams( $frame ) );
```

Comment:
:+1: for extracting this method. Reads much easier now at this level of abstraction